pkg/core/blockchain.go

```diff
@@ -647,12 +656,37 @@ func (bc *Blockchain) Run() {
 		case <-bc.stopCh:
 			return
 		case <-persistTimer.C:
+			var oldPersisted uint32
+			var gcDur time.Duration
+
+			if bc.config.RemoveUntraceableBlocks {
+				oldPersisted = atomic.LoadUint32(&bc.persistedHeight) / bc.config.GarbageCollectionPeriod
+			}
 			dur, err := bc.persist(nextSync)
 			if err != nil {
 				bc.log.Warn("failed to persist blockchain", zap.Error(err))
 			}
+			if bc.config.RemoveUntraceableBlocks {
+				newPersisted := atomic.LoadUint32(&bc.persistedHeight) / bc.config.GarbageCollectionPeriod
+				var tgtBlock = int64(newPersisted)
+
+				tgtBlock -= int64(bc.config.MaxTraceableBlocks)
```

Comment:
It seems like `tgtBlock` can become negative here. Do we still behave correctly in this case?

Reply:
I think so. Though it was tested im a bit different setup.

Reply:
Oh, missed that `GC` actually runs only if `tgtBlock > int64(bc.config.GarbageCollectionPeriod)`, so it should be file.